runner/remote.go

```diff
@@ -0,0 +1,417 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/kr/pty"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+var (
+	blackListed1stLevel = []string{"admin", "cli", "config", "core", "db", "dist-archive",
+		"eval-file", "eval", "find", "i18n", "scaffold", "server", "package", "profile"}
+
+	blackListed2ndLevel = map[string][]string{
+		"media":    {"regenerate"},
+		"theme":    {"install", "update", "delete"},
+		"plugin":   {"install", "update", "delete"},
+		"language": {"install", "update", "delete"},
+		"vip":      {"support-user"},
+	}
+
+	guidRegex *regexp.Regexp
+	padlock   *sync.Mutex
+)
+
+func waitForConnect() {
+	padlock = &sync.Mutex{}
+	guidRegex = regexp.MustCompile("^[a-fA-F0-9\\-]+$")
+	if nil == guidRegex {
+		logger.Println("Failed to compile the Guid regex")
+		return
+	}
+
+	addr, err := net.ResolveTCPAddr("tcp4", "0.0.0.0:22122")
+	if err != nil {
+		logger.Printf("error resolving listen address: %s\n", err.Error())
+		return
+	}
+
+	listener, err := net.ListenTCP("tcp4", addr)
+	if err != nil {
+		logger.Printf("error listening on %s: %s\n", addr.String(), err.Error())
+		return
+	}
+	defer listener.Close()
+
+	for {
+		logger.Println("listening...")
+		conn, err := listener.AcceptTCP()
+		logger.Println("connection: ", conn.LocalAddr().String())
+		if err != nil {
+			logger.Printf("error accepting connection: %s\n", err.Error())
+			continue
+		}
+		go authConn(conn)
+	}
+}
+
+func authConn(conn *net.TCPConn) {
+	elems := make([]string, 0)
+	var rows, cols uint64
+	var Guid string
+
+	for {
+		data := make([]byte, 256)
```

Comment:
Noting that the `256` here limits the length of the wp-cli command to around ~150 chars

Reply:
Switched to using bufio, so the effective total packet size is now 64k.